### PR TITLE
Issue 57 oauth documentation

### PR DESCRIPTION
**Pull request by dragonprevost**
This pull request has implemented functions to execute the first steps to Oauth. The login process has not changed for the user, but the functions to implement Oauth are there for the following sprint. 
### Changes
* electron-oauth2 package added to repo
* Functions for storing access token in file system implemented in `app/misc/storeCredentials.ts`
* Instance of Oauth object created in `app/misc/authenticate.ts`
* Function to trigger Oauth window, and token retrieval implemented in `app/misc/authenticate.ts`
* (not in PR) Wiki created documenting to process of Oauth with github and how it integrates with VisualGit

### CODE REVIEW

File: app/misc/storeCredentials.ts [hunk 1]
```diff
@@ -19,6 +19,27 @@ function encrypt(username, password) {
...
+  var obj = {'accessToken': encryptedAccessToken.toString()};
+
+  jsonfile.writeFile(file, obj, function (err) {
+    if (err) throw err;
+    console.log('Access Token Saved to File System');
+
+  })
+}
+
+function getAccesToken(){
```
Comment (sterlinglaird) [inline app/misc/storeCredentials.ts]: typo

File: app/misc/authenticate.ts [hunk 2]
```diff
@@ -25,8 +27,26 @@ let loginScopes = [
   "user"
 ];
 
-//Called then user pushes to sign out even if they have commited changes but not pushed; prompts a confirmation modal
+const OauthConfig = {
+  clientId: 'CLIENT_ID',
```
[the remark this answers is not in the document]
Comment (dragonprevost) [reply app/misc/authenticate.ts]: I will have to investigate the best solution for storing client secrets. For now I believe, as long as we are not pushing the secret to github we should be alright.

File: app/misc/storeCredentials.ts (hunk 1 above)
Comment (dragonprevost) [reply app/misc/storeCredentials.ts]: Good catch! Thanks.

**Comment by colinwerner** [comment]
May I suggest we do not merge this until after our demo? (As for the demo, I'd like to ask @dragonprevost to talk about the research as part of the demo?)

**Comment by dragonprevost** [comment]
Hey @colinwerner, We certainly can hold off until the next sprint, but I believe this pull request could be a good demonstration of how our sprint points are being expensed. It will not affect the user login process in the app but the code is organized and will be used in the following sprint. 

What do you think?

Also I am happy to contribute our Oauth findings in the demo.

**Comment by colinwerner** [comment]
> Hey @colinwerner, We certainly can hold off until the next sprint, but I believe this pull request could be a good demonstration of how our sprint points are being expensed. It will not affect the user login process in the app but the code is organized and will be used in the following sprint.
> 
> What do you think?
> 
> Also I am happy to contribute our Oauth findings in the demo.

I totally agree it does give us some velocity to our sprint. I have two concerns:
1.) the demo is tomorrow
2.) we have two other changes that need to be merged

In a perfect world there would be no merge conflicts and everything will work as planned. However, seeing as though this is our first sprint I want to make sure it goes smoothly. Therein lies the conflict... I'm open to discussing further; however, we don't have a lot of time left.

**Comment by dragonprevost** [comment]
> > Hey @colinwerner, We certainly can hold off until the next sprint, but I believe this pull request could be a good demonstration of how our sprint points are being expensed. It will not affect the user login process in the app but the code is organized and will be used in the following sprint.
> > What do you think?
> > Also I am happy to contribute our Oauth findings in the demo.
> 
> I totally agree it does give us some velocity to our sprint. I have two concerns:
> 1.) the demo is tomorrow
> 2.) we have two other changes that need to be merged
> 
> In a perfect world there would be no merge conflicts and everything will work as planned. However, seeing as though this is our first sprint I want to make sure it goes smoothly. Therein lies the conflict... I'm open to discussing further; however, we don't have a lot of time left.

Yeah I agree this is last minute to be merging new branches into master. Since none of the implemented functions are in use, we really should not encounter any bugs. Also the PR is flagged as having no conflicts with the base branch. So we should be able just merge it without needing to resolve any merge conflicts.

Let me know what you think.

**Comment by colinwerner** [comment]
Feel free to merge it -- but please do a quick test to ensure all the functionality that will be demoed tomorrow works as expected, please.

**Pull request merged**